lib/_007/Runtime.pm6

```diff
@@ -435,9 +435,15 @@ class _007::Runtime {
         }
         elsif $obj ~~ Val::Str && $propname eq "substr" {
             return builtin(sub substr($pos, $chars) {
-                return Val::Str.new(:value($obj.value.substr(
-                    $pos.value,
-                    $chars.value)));
+                my $s = $obj.value;
+
+                die X::Subscript::TooLarge.new(:value($pos.value), :length($s.chars))
+                    if $pos.value >= $s.chars + 1;
```

Comment:
Why >= + 1 ?

Reply:
Good catch! Not wrong, though.

This is a lesser-known but completely deliberate quirk of `substr`. I just re-confirmed it in both Perl 5 `perl` and Rakudo Perl 6. (The former returns `undef`; the latter throws an exception.) You're allowed to start the `substr` operation on index `length-of-string`, but not on index `length-of-string + 1`.